Switched to Drupal_TI

LKS90 opened the pull request:
Testing redirect with Drupal_TI

fabianx-ai commented on .travis.yml:
```diff
@@ -1,102 +1,113 @@
+    #- DRUPAL_TI_RUNNERS="phpunit"
+    #- DRUPAL_TI_RUNNERS="simpletest"
+    #- DRUPAL_TI_RUNNERS="behat"
+    - DRUPAL_TI_RUNNERS="phpunit-core simpletest"
```
Move that to env, and move SUBDIR back in here.

fabianx-ai commented on .travis.yml:
```diff
@@ -1,102 +1,113 @@
+    - DRUPAL_TI_WEBSERVER_PORT="8080"
+
+    # Simpletest specific commandline arguments, the DRUPAL_TI_SIMPLETEST_GROUP is appended at the end.
+    - DRUPAL_TI_SIMPLETEST_ARGS="--verbose --color --concurrency 4 --url $DRUPAL_TI_WEBSERVER_URL:$DRUPAL_TI_WEBSERVER_PORT"
```
Add $DRUPAL_TI_WEBSERVER_PORT/$SUBDIR here.

fabianx-ai commented on .travis.yml:
```diff
@@ -1,102 +1,113 @@
 
 env:
-  - SUBDIR=../drupal URL=http://localhost
-  - SUBDIR=.. URL=http://localhost/drupal
```
Define subdir as once SUBDIR="" and once SUBDIR="subdir"

fabianx-ai commented on .travis.yml:
```diff
@@ -1,102 +1,113 @@
-  - drush en --yes simpletest
-  - drush cr
-  - phpenv rehash
+  - drupal-ti before_script
```
Add:
- drupal-ti --include .before-script.sh

And in the .before-script.sh do:

cd $DRUPAL_TI_DRUPAL_DIR
ln -sf $(pwd) subdir

fabianx-ai commented on .travis.yml:
```diff
@@ -1,102 +1,118 @@
+    - DRUPAL_TI_WEBSERVER_PORT="8080"
+
+    # Simpletest specific commandline arguments, the DRUPAL_TI_SIMPLETEST_GROUP is appended at the end.
+    - DRUPAL_TI_SIMPLETEST_ARGS="--verbose --color --concurrency 4 --url $DRUPAL_TI_WEBSERVER_URL:$DRUPAL_TI_WEBSERVER_PORT/$SUBDIR"
```
Almost there, that won't work as the matrix comes after the globals, so this is not having an effect.

Maybe move try moving the matrix: above the global: section.

No idea how travis parses it though.

fabianx-ai replied:
Comment out this line and add:

```
- export DRUPAL_TI_SIMPLETEST_ARGS="--verbose --color --concurrency 4 --url $DRUPAL_TI_WEBSERVER_URL:$DRUPAL_TI_WEBSERVER_PORT/$SUBDIR"
```

as first line in the script section.

fabianx-ai commented on .travis-before-script.sh:
```diff
@@ -0,0 +1,15 @@
+
+# These variables come from environments/drupal-*.sh
+mkdir -p "$DRUPAL_TI_MODULES_PATH"
+cd "$DRUPAL_TI_MODULES_PATH"
```
This is likely not needed in this project.

fabianx-ai commented on .travis-before-script.sh:
```diff
@@ -0,0 +1,15 @@
+
+# Add needed dependencies.
+cd "$DRUPAL_TI_DRUPAL_DIR"
+ln -sf $(pwd) subdir
```
Uhm, probably it would make more sense to use a .travis-after-script.sh and then have this run after drupal installation, but as "drupal_ti_ensure_drupal" is re-rentrant, it does not matter much either way.

Berdir commented:
Yep, the matrix isn't working yet: https://travis-ci.org/md-systems/redirect/builds/74225970

The expected output is to have two test runs per php version.

fabianx-ai commented:
@Berdir https://travis-ci.org/md-systems/redirect/builds/74234438 the matrix works fine, it is just that the test runner is not using the subdir, because of:

export DRUPAL_TI_SIMPLETEST_ARGS=".../$SUBDIR"

then

export SUBDIR="..."

So a typical dependency problem.

I am not sure if the matrix always comes first, in which case the simplest is to comment out the env directive and put the export statement as one of the first commands instead.

If the matrix comes for whatever global / matrix comes first, then could also try switching around.

LKS90 commented:
I think it's working now, two test runs per php version, once the subdir is empty, once it's subdir. Moving the matrix above the env was enough. I also moved the linking to $SUBDIR to an after-script and removed the unnecessary command from the before script.

fabianx-ai commented on .travis-before-script.sh:
```diff
@@ -0,0 +1,7 @@
+
+# Ensure the right Drupal version is installed.
+# Note: This function is re-entrant.
+drupal_ti_ensure_drupal
```
This script can be completely removed now as all of this is also done by the simpletest runners script.sh

fabianx-ai commented:
No, unfortunately the subdir is still put last.

I'll leave some detailed comments.

fabianx-ai commented on .travis.yml:
```diff
@@ -1,102 +1,118 @@
+  - drupal-ti script
+
+after_script:
+  - drupal-ti --include .travis-after-script.sh
```
Uhm, this was meant to be called like:

```
.travis-after-before_script.sh
```

And then be called in the before_script section after the before_script command.

Likely could also call it in the script section before the `drupal-ti script` though.

which probably is best.

fabianx-ai commented on .travis.yml:
```diff
@@ -1,102 +1,118 @@
-  - sudo cat /var/log/apache2/error.log
-  # Exit the build
-  - if [ $TEST_EXIT -eq 0 ] && [ $TEST_OUTPUT -eq 0 ] && [ $TEST_PHPUNIT -eq 0 ]; then exit 0; else exit 1; fi
+  - drupal-ti script
```
Yeah, lets move the include just above this line.

Berdir commented:
Nice, passes now that we fixed the test. Merged.
